src/CMakeLists.txt

```diff
@@ -26,8 +26,9 @@ add_library(${PROJECT_NAME} ${SOURCES})
 target_include_directories(${PROJECT_NAME}
                            PRIVATE
                            ${PROJECT_SOURCE_DIR}/include
-                           ${PROJECT_BINARY_DIR}/include
-                           PUBLIC
```

Comment:
Do you remember what's the rationale here?

Reply:
When include dirs of external libraries are added via `target_include_directories(SYSTEM)`, CMake passes them to a compiler using the `-isystem` command line flag instead of `-I`. Headers added this way get a special treatment. In particular, the compiler will not generate warnings in such headers. Those warnings would be a useless distraction as there is normally no way to modify code in the external libraries.